src/install/install.zig

```diff
@@ -7688,8 +7688,21 @@ pub const PackageManager = struct {
             }
 
             const trees = this.lockfile.buffers.trees.items;
-            this.tree_install_counts[tree_id] += 1;
-            if (this.tree_install_counts[tree_id] >= trees[tree_id].dependencies.len) {
+            const current_count = this.tree_install_counts[tree_id];
+            const max = trees[tree_id].dependencies.len;
+
+            if (current_count == std.math.maxInt(usize)) {
```

Comment:
@dylan-conway the previous assertion for this was failing in the `test/` folder in bun's repo when `rm -rf ~/.bun/install/cache`

I suspect the count included workspace scripts which may have not completed yet